go/common/crypto/signature/signers/plugin/plugin.go

```diff
@@ -77,26 +77,9 @@ func NewFactory(config interface{}, roles ...signature.SignerRole) (signature.Si
 		return nil, fmt.Errorf("signature/signer/plugin: a plugin name must be specified")
 	}
 
-	// Why yes, the correct thing to do is to call `client.Kill`,
-	// when we are done with the plugin.  Alas there's no good
-	// way to do that while using `os.Exit`.
-	//
-	// At least ensure that child processes get killed in sensible
-	// environments.  If people have a problem with this they can
-	// do one or more of:
-	//
-	//  1. Get rid of the ~180 instances of `os.Exit`
-	//  2. Complain to the Go developers to make `plugin` not suck.
-	//  3. Complain to their OS vendor to implement PR_SET_PDEATH_SIG`.
-	//  4. Complain to Docker to make Docker not suck (reaping zombies
-	//     isn't that hard).
-	//
-	// We do use managed plugins so that in the graceful exit case,
-	// the cleanup will be done at least.
```

Comment:
Why did you remove this, the comment still applies, for Mac OS specifically:
> //  3. Complain to their OS vendor to implement PR_SET_PDEATH_SIG`.

:grin:

Reply:
Yeah, all of this comment is still applicable.

Reply:
Restored 😄